In [1]:
import pandas as pd
import numpy as np

In [2]:
df=pd.read_csv("https://catalog.ourworldindata.org/garden/covid/latest/compact/compact.csv")

In [69]:
df.head(5)

,country,date,total_cases,new_cases,new_cases_smoothed,total_cases_per_million,new_cases_per_million,new_cases_smoothed_per_million,total_deaths,new_deaths,...,population,population_density,median_age,life_expectancy,gdp_per_capita,extreme_poverty,diabetes_prevalence,handwashing_facilities,hospital_beds_per_thousand,human_development_index
0,Afghanistan,2020-01-01,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,40578847.0,62.215549,16.752001,65.616997,1983.812622,NaN,11.7,51.938343,0.35,NaN
1,Afghanistan,2020-01-02,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,40578847.0,62.215549,16.752001,65.616997,1983.812622,NaN,11.7,51.938343,0.35,NaN
2,Afghanistan,2020-01-03,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,40578847.0,62.215549,16.752001,65.616997,1983.812622,NaN,11.7,51.938343,0.35,NaN
3,Afghanistan,2020-01-04,0.0,0.0,NaN,0.0,0.0,NaN,0.0,0.0,...,40578847.0,62.215549,16.752001,65.616997,1983.812622,NaN,11.7,51.938343,0.35,NaN
4,Afghanistan,2020-01-05,0.0,0.0,NaN,0.0,0.0,NaN,0.0,0.0,...,40578847.0,62.215549,16.752001,65.616997,1983.812622,NaN,11.7,51.938343,0.35,NaN


In [3]:
df.memory_usage(deep=True).sum()/1024**2

np.float64(382.2965841293335)

In [70]:
last_date_of_record=df["date"].max()   
first_date_of_record=df["date"].min()
no_of_countries = df["country"].nunique()
country_count=df["country"].value_counts()
Countries=country_count[country_count==2322].index

**->Checking For the last_date AND first_date such that having a idea about when the data is being analyzed what was the recent data**

**->Also check the no. of countries in the data set to get idea about how much no. of contry's analysis is done**

**->On checking i understood that there is some countries or events where we dont have this much complete end to end data so we seperated out those data**

In [71]:
df_complete=df[df["country"].isin(Countries)]
df_complete["country"].value_counts()

country
Afghanistan    2322
Albania        2322
Algeria        2322
Andorra        2322
Angola         2322
               ... 
Venezuela      2322
Vietnam        2322
Yemen          2322
Zambia         2322
Zimbabwe       2322
Name: count, Length: 181, dtype: int64

*->Got that total number of contry which is 181 who have end to end analysis right from first date to last date*

In [84]:
missing_counts=df.groupby("country")["total_cases"].apply(lambda x: x.isnull().sum())
missing_counts[missing_counts==1672]

country
Hong Kong    1672
Name: total_cases, dtype: int64

*-> Got total number of null values in total_Cases reported maximum in hongkong as 1672 values*

In [105]:
total_cases_max = df.groupby("country")["total_cases"].max()

total_cases_max.sort_values(ascending=False).head(15)

country
World                                                  779131743.0
World excl. China                                      679749982.0
World excl. China and South Korea                      645178109.0
World excl. China, South Korea, Japan and Singapore    608368382.0
High-income countries                                  458049345.0
Asia                                                   302181607.0
Europe                                                 254960190.0
Upper-middle-income countries                          239829083.0
Asia excl. China                                       202799846.0
European Union (27)                                    186985473.0
North America                                          124520038.0
United States                                          103436829.0
China                                                   99381761.0
Lower-middle-income countries                           78235498.0
South America                                         

In [106]:
new_cases_max=df.groupby("country")["new_cases"].max()
new_cases_max.sort_values(ascending=False).head(10)

country
World                                                  8401906.0
World excl. China                                      8401713.0
World excl. China and South Korea                      8384638.0
World excl. China, South Korea, Japan and Singapore    8295098.0
Asia                                                   7213802.0
High-income countries                                  7197333.0
Upper-middle-income countries                          7027032.0
China                                                  6966046.0
Europe                                                 6386449.0
European Union (27)                                    5896096.0
Name: new_cases, dtype: float64

In [107]:
df["date"][df["new_cases"]==6966046]

105547    2022-12-23
Name: date, dtype: object

->Highest case in a day reported by any Individual country is China and which is On 2022-12-2023

In [109]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 589779 entries, 0 to 589778
Data columns (total 61 columns):
 #   Column                                      Non-Null Count   Dtype  
---  ------                                      --------------   -----  
 0   country                                     589779 non-null  object 
 1   date                                        589779 non-null  object 
 2   total_cases                                 577431 non-null  float64
 3   new_cases                                   573183 non-null  float64
 4   new_cases_smoothed                          571974 non-null  float64
 5   total_cases_per_million                     577431 non-null  float64
 6   new_cases_per_million                       573183 non-null  float64
 7   new_cases_smoothed_per_million              571974 non-null  float64
 8   total_deaths                                577431 non-null  float64
 9   new_deaths                                  574285 non-null  float64
 

In [10]:
for cols in df.columns:
    print(cols)

country
date
total_cases
new_cases
new_cases_smoothed
total_cases_per_million
new_cases_per_million
new_cases_smoothed_per_million
total_deaths
new_deaths
new_deaths_smoothed
total_deaths_per_million
new_deaths_per_million
new_deaths_smoothed_per_million
excess_mortality
excess_mortality_cumulative
excess_mortality_cumulative_absolute
excess_mortality_cumulative_per_million
hosp_patients
hosp_patients_per_million
weekly_hosp_admissions
weekly_hosp_admissions_per_million
icu_patients
icu_patients_per_million
weekly_icu_admissions
weekly_icu_admissions_per_million
stringency_index
reproduction_rate
total_tests
new_tests
total_tests_per_thousand
new_tests_per_thousand
new_tests_smoothed
new_tests_smoothed_per_thousand
positive_rate
tests_per_case
total_vaccinations
people_vaccinated
people_fully_vaccinated
total_boosters
new_vaccinations
new_vaccinations_smoothed
total_vaccinations_per_hundred
people_vaccinated_per_hundred
people_fully_vaccinated_per_hundred
total_boosters_per_hundred
new